src/header-generator.ts

```diff
@@ -133,6 +52,31 @@ function getOrderFromUserAgent(headers) {
  *  Either 1 or 2. If none specified the httpVersion specified in `HeaderGeneratorOptions` is used.
  */
 
+export interface BrowserSpecification {
+    name: typeof SUPPORTED_BROWSERS[number];
+    minVersion?: number;
+    maxVersion?: number;
+    httpVersion?: typeof SUPPORTED_HTTP_VERSIONS[number];
+}
+
+export type BrowsersType = BrowserSpecification[] | (typeof SUPPORTED_BROWSERS[number])[];
+
+export interface HeaderGeneratorOptions {
+    browsers: BrowsersType;
+    browserListQuery: string;
+    operatingSystems: (typeof SUPPORTED_OPERATING_SYSTEMS[number])[];
+    devices: (typeof SUPPORTED_DEVICES[number])[];
+    locales: string[];
+    httpVersion: typeof SUPPORTED_HTTP_VERSIONS[number];
+}
+
+export type HttpBrowserObject = {
+    name: BrowserSpecification['name'] | typeof MISSING_VALUE_DATASET_TOKEN,
```

Comment:
i would prefer to stick to `;` as a separator in types (both interfaces and types do support both `,` and `;`)

this can be enforced via eslint, will be probably add that to the apify eslint ts config soon anyway